src/framework/ui/internal/uiconfiguration.cpp

```diff
@@ -173,3 +179,58 @@ float UiConfiguration::physicalDotsPerInch() const
     }
     return 100;
 }
+
+QByteArray UiConfiguration::pageState(const std::string& pageName) const
+{
+    TRACEFUNC;
+    std::string stateString = uiValue(STATES_PATH + "/" + pageName).toString();
+    return stringToByteArray(stateString);
+}
+
+void UiConfiguration::setPageState(const std::string& pageName, const QByteArray& state)
+{
+    TRACEFUNC;
+    setUiValue(STATES_PATH + "/" + pageName, Val(byteArrayToString(state)));
+}
+
+Notification UiConfiguration::pageStateChanged() const
+{
+    return m_pageStateChanged;
+}
+
+QByteArray UiConfiguration::stringToByteArray(const std::string& string) const
+{
+    QString qString = QString::fromStdString(string);
+    QByteArray byteArray64(qString.toUtf8());
+    QByteArray byteArray = QByteArray::fromBase64(byteArray64);
+
+    return byteArray;
+}
+
+std::string UiConfiguration::byteArrayToString(const QByteArray& byteArray) const
+{
+    QByteArray byteArray64 = byteArray.toBase64();
+    return byteArray64.toStdString();
+}
+
+mu::Val UiConfiguration::uiValue(const std::string& key) const
+{
+    Val val = workspaceUiArrangment()->value(key);
+    if (!val.isNull()) {
```

Comment:
Better to add contains(key) to an interface of this class and hide this logic inside it